addons/stock/wizard/stock_return_picking.py

```diff
@@ -25,6 +25,9 @@ class stock_return_picking(osv.osv_memory):
     _columns = {
         'product_return_moves': fields.one2many('stock.return.picking.line', 'wizard_id', 'Moves'),
         'move_dest_exists': fields.boolean('Chained Move Exists', readonly=True, help="Technical field used to hide help tooltip if not needed"),
+        'parent_location_id': fields.many2one('stock.location'),
+        'location_id': fields.many2one('stock.location', 'Return Location',
+            domain="[('return_location', '=', True), ('id', 'child_of', parent_location_id)]")
```

Comment:
Ok, I understand the idea of having a location in the same warehouse, but your domain should also return to the original location even if it is not a return location or not in the same warehouse

['|', ('id', '=', originallocation), '&' ('return_locationTrue), (id, child_of..)]